content/source/docs/enterprise-beta/api/modules.html.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: enterprise2
+page_title: "Modules - API Docs - Terraform Enterprise Beta"
+sidebar_current: "docs-enterprise2-api-modules"
+---
+
+# Registry Modules API
+
+-> **Note**: These API endpoints are in beta and are subject to change.
+
+## List modules
+
+Lists all the modules for a given organization. **Note** that all other endpoints have the subpath `/api/v2`, this endpoint does not use that subpath.
```

Comment:
Hmm this ends up being kinda janky.

The real reason is that this is actually part of the Public API although we didn't actually document then "browse a single namespace" option yet. It's hard to know how best to explaina that.

I'd be happy to document this in public API and then explain here that this is part of public API and that TFE maps `namespace` to `organisation` name.

What do you think?

Reply:
We have this: https://www.terraform.io/docs/registry/api.html#list-latest-version-of-module-for-all-providers but basically need to add the version without the module name too then cross link this to there and explain that it's the same and response format/pagination is documented there etc.

Reply:
@mitchellh and I agreed we should just document this endpoint in the public docs and then link there like with the other public endpoints.

I started looking at that and realise it's actually a little bit hairy since right now we don't list _any_ of the browse or search endpoints in public - it was a TODO item. Im going to fix that up so we can cross link but now also hitting issues with the way we specified Discovery documents - they currently include trailing slash in the "prefix" however the actual API for listing modules 301 redirects that back to no slash which is pretty gross to try an document.

I'll think about it while I'm travelling next week and either test out and change the discovery documents to not include trailing (we think TF CLI will handle that already), or I'll leave that but change the public API a little and update the registry and TFE so they aren't relying on 301 redirects.

Kicking myself for not spotting this silly little one character that makes this all messy earlier.

**tl;dr** you should probably not bother documenting this here and link to Public API (https://www.terraform.io/docs/registry/api.html) instead for all read operations with a note about what the path prefix is on TFE maybe?

Reply:
I just gutted this section and replaced it with the reference to the Public API docs with details on the deltas and an example.